### PR TITLE
Feat/inject commit hash

### PR DESCRIPTION
**Pull request by MasterKenth**
## Explain the changes you’ve made

Injected the current git commit hash into the app so that it is available at runtime.

## Explain why these changes are made

Hash will be used to display version info in a coming update.

## Explain your solution

A bash script has been added that runs on yarn postinstall and before ios/android/start. This should ensure that the hash is always up-to-date both when developing and when building.

The bash script adds a `GIT_COMMIT_HASH` variable to the .env file. If the variable already exists in the file then it is replaced to avoid duplicates.

## How to test

Concrete example:

1. Checkout this branch
2. Run any command like `yarn install` or `yarn start` and verify that `GIT_COMMIT_HASH` has been added to your `.env` file.
3. Change the value in `.env`, run any command again and verify that your change is overwritten and `GIT_COMMIT_HASH` is updated correctly.
4. Run the app and verify that `GIT_COMMIT_HASH` is available and correct. Use something like:

```javascript
import Config from "react-native-config";

...

console.log("GIT_COMMIT_HASH = ", Config.GIT_COMMIT_HASH);
```

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [x] Building the Application on a Android device/simulator.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/fedfce3fd8695f238dea9cb00bd37ce7/pull-request/594)

File: scripts/add-commit-hash-to-env.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+if [ -f ".env" ]
```
Comment (karlknowit) [inline scripts/add-commit-hash-to-env.sh:1]: Aren't you missing `#!/bin/bash` in top of the file?

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: scripts/add-commit-hash-to-env.sh (hunk 1 above)
Comment (MasterKenth) [reply scripts/add-commit-hash-to-env.sh:1]: Yep, will fix!

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1kjdu4b Injicera commit hash innan builds](https://app.clickup.com/t/1kjdu4b)

**Review by dannil76 — APPROVED** [review]

**Pull request merged**